src/DynamoCore/Engine/EngineController.cs

```diff
@@ -518,8 +513,18 @@ private void OnLibraryLoaded()
         /// </summary>
         private void LibraryLoaded(object sender, LibraryServices.LibraryLoadedEventArgs e)
         {
-            if(e.LibraryPaths.Any())
+            if (e.LibraryPaths.Any())
+            {
+                // Need to make compiled custom nodes available before resetting the VM
+                OnRequestCustomNodeRegistration();
                 OnLibraryLoaded();
+            }
+        }
+
+        internal event EventHandler RequestCustomNodeRegistration;
```

Comment:
Can this be a normal delegate as it doesn't require any arguments that are required by the `EventHandler` type?

Reply:
You are right, but on the other hand events can support multiple handlers if needed. If you feel strongly about this I can still change it, probably multiple handlers will never be needed.

Reply:
It's minor, I'm not really particular about it, just a thought.